sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -794,11 +794,11 @@ public void process(ProcessContext c) throws Exception {
                 ? writeOperation.finalizeDestination(
                     defaultDest, GlobalWindow.INSTANCE, fixedNumShards, fileResults)
                 : finalizeAllDestinations(fileResults, fixedNumShards);
+        writeOperation.moveToOutputFiles(resultsToFinalFilenames);
```

Comment:
This looks good to me (and I'll go ahead and merge this PR). However I am confused as to why this makes a difference. My understanding is that the outputs should not be delivered until after the DoFn finishes executing, which means that the order inside the DoFn should not matter.

Reply:
Strange. I do not know about internals of beam soI can not comment. But empirical observation is, that my code is not runnable without this patch. I am running Beam 2.5.0 on Flink 1.4.0.